WooCommerce/Classes/ViewRelated/Dashboard/Factories/DashboardUIFactory.swift

```diff
@@ -103,3 +112,16 @@ private extension DashboardUIFactory {
         }
     }
 }
+
+private extension Date {
+
+    /// Returns a date object that corresponds to September 1st of 2020
+    static var september1st2020: Date? {
+        var dateComponents = DateComponents()
+        dateComponents.year = 2020
```

Comment:
my test device happens to be on Japanese calendar (under Settings > General > Language & Region > Calendar), and I just couldn't get the post Sep 1 UI to show while passing date on Sep 1 in my local time. then I checked `september1st2020`, and it showed `4038-08-31 16:00:00 +0000` 😮  then I realized it might be due to the calendar, and got to reproduce on a simulator with a Japanese calendar. I also tried a temp test case and it passes with Gregorian calendar but not Japanese calendar:

```
func testDeprecatedDate() {
        var dateComponents = DateComponents()
        dateComponents.year = 2020
        dateComponents.month = 9
        dateComponents.day = 1

        let deprecatedDate = Calendar.current.date(from: dateComponents)!

        // Tuesday, September 1, 2020 5:39:20 AM GMT+08:00
        let laterDate = Date(timeIntervalSince1970: 1598909960)
        XCTAssertLessThanOrEqual(deprecatedDate, laterDate)
}
```

maybe with a year (2020) set in Gregorian calendar system, we want to also specify to use Gregorian calendar in line 125?

Reply:
Wow, very nice catch!
I've made changes to specify a gregorian calendar since we are using a gregorian year! https://github.com/woocommerce/woocommerce-ios/pull/2521/commits/878960412b42275fdee841810ff6284b0c0de401

Reply:
yayy working now! 🎉